core-gen/src/test/scala/org/bitcoins/core/gen/ln/LnInvoiceGen.scala

```diff
@@ -0,0 +1,129 @@
+package org.bitcoins.core.gen.ln
+
+import org.bitcoins.core.gen._
+import org.bitcoins.core.number.{ UInt64, UInt8 }
+import org.bitcoins.core.protocol.ln._
+import org.bitcoins.core.protocol.ln.routing.LnRoute
+import org.bitcoins.core.util.NumberUtil
+import org.scalacheck.Gen
+
+sealed abstract class LnInvoiceGen {
+
+  /**
+   * Generates a [[org.bitcoins.core.protocol.ln.LnHumanReadablePart]]
+   * that does not contain a amount
+   * @return
+   */
+  def lnHrpNoAmt: Gen[LnHumanReadablePart] = {
+    ChainParamsGenerator.lnNetworkParams.flatMap {
```

Comment:
Is there any difference in doing `flatMap` vs `map` here?

Reply:
Yes. If we did not use `.flatMap` here the resulting type would be `Gen[Gen[LnHumanReadablePart]]`. 

The use of `.flatMap` here collapses the `Gen[Gen[]]` into just one `Gen[]`. 

You will see this pattern used consitently across Scala code bases, for instance the same applies to `Future[Future[T]]`. 

```scala
//since we use `.map` here get nested futures
val x: Future[Future[Int]] = Future.successful(1).map(_ => Future.successful(2))

val y: Future[Int] = Future.successful(1).flatMap(_ => Future.successful(2))
```